src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -0,0 +1,75 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def load_command_table(self, _):
+    from azure.cli.core.commands import CliCommandType
+    from azure.cli.core.profiles import ResourceType
+    from ._client_factory import cf_synapse_client_workspace_factory
+    from ._client_factory import cf_synapse_client_bigdatapool_factory
+    from ._client_factory import cf_synapse_spark_batch
+    from ._client_factory import cf_synapse_spark_session
+
+    synapse_workspace_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.synapse.operations#WorkspacesOperations.{}',
+        client_factory=cf_synapse_client_workspace_factory,
+        resource_type=ResourceType.MGMT_SYNAPSE
+    )
+
+    synapse_bigdatapool_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.synapse.operations#BigDataPoolsOperations.{}',
+        client_factory=cf_synapse_client_bigdatapool_factory,
+        resource_type=ResourceType.MGMT_SYNAPSE
+    )
+
+    synapse_spark_batch_sdk = CliCommandType(
+        operations_tmpl='azure.synapse.operations#SparkBatchOperations.{}',
+        client_factory=cf_synapse_spark_batch
+    )
+
+    synapse_spark_session_sdk = CliCommandType(
+        operations_tmpl='azure.synapse.operations#SparkSessionOperations.{}',
+        client_factory=cf_synapse_spark_session
+    )
+
+    # Management Plane Commands
+    with self.command_group('synapse workspace', synapse_workspace_sdk,
+                            client_factory=cf_synapse_client_workspace_factory) as g:
+        g.show_command('show', 'get')
+        g.custom_command('list', 'list_workspaces')
+        g.custom_command('create', 'create_workspace', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+
+    with self.command_group('synapse spark pool', synapse_bigdatapool_sdk,
+                            client_factory=cf_synapse_client_bigdatapool_factory) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_workspace')
+        g.custom_command('create', 'create_spark_pool', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+
+    # Data Plane Commands
+    # Spark batch opertions
+    with self.command_group('synapse spark batch', synapse_spark_batch_sdk, client_factory=cf_synapse_spark_batch) as g:
+        g.custom_command('create', 'create_spark_batch_job', supports_no_wait=True)
+        g.custom_command('list', 'list_spark_batch_jobs')
+        g.custom_command('show', 'get_spark_batch_job')
+        g.custom_command('cancel', 'delete_spark_batch_job', confirmation=True, supports_no_wait=True)
```

Comment:
better to use "delete" instead of "cancel" [https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior)

Reply:
@jsntcy , I think `cancel` would be a better verb for job. Actually, we used `delete` in our draft version and our partner team provides a suggestion to rename it as `cancel`. The issue here is we should also rename the SDK method to `cancel_spark_batch_job`. @aim-for-better , could you please create a work item to track this issue in our internal Azure Devops project?

Reply:
> better to use "delete" instead of "cancel" https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior

This command is for data plane. It will cancel a job instead of deleting a resource. Our internal  customer think that cancel has a better semantic than delete. I have changed `delete_spark_batch_job` to `cancel_spark_batch_job`

Reply:
> @jsntcy , I think `cancel` would be a better verb for job. Actually, we used `delete` in our draft version and our partner team provides a suggestion to rename it as `cancel`. The issue here is we should also rename the SDK method to `cancel_spark_batch_job`. @aim-for-better , could you please create a work item to track this issue in our internal Azure Devops project?

Have change the method name to`cancel_spark_batch_job`